### PR TITLE
added Property Filter plugin

### PR DESCRIPTION
**Pull request by lsmith77**

### CODE REVIEW

File: plugins.json [hunk 1]
```diff
@@ -343,5 +343,14 @@
     "maintainer": "official",
     "displayOnWebsiteLib": true,
     "type": "data_in"
+  },
+  {
+    "name": "Property Filter",
+    "url": "  https://github.com/witty-works/posthog-property-filter-plugin",
```
Comment (yakkomajuri) [inline plugins.json:349]:
```suggestion
    "url": "https://github.com/witty-works/posthog-property-filter-plugin",
```

**Review by yakkomajuri — COMMENTED** [review]
A few questions about the plugin code:

```js
async function setupPlugin({ config, global }) {
    // nit: given we're just going to iterate over this a set isn't necessary
    global.propertiesToFilter = new Set(config.properties.split(','))
}

async function processEvent(event, { global, storage }) {
    // this seems like some very specific handling to include in a generic plugin
    if (event.event === 'session_started') {
        return event
    }

    global.propertiesToFilter.forEach(async function (propertyToFilter) {
        // I'm slightly wary of introducing something that modifies the actual event payload but maybe you need this for ip? perhaps we can have this be very specific to IP only to avoid unintended consequences?
        if (propertyToFilter in event) {
            delete event[propertyToFilter]
        }

        if (event.properties && propertyToFilter in event.properties) {
            delete event.properties[propertyToFilter]
        }
    })

    return event
}
```

**Comment by yakkomajuri** [comment]
Thanks for doing this btw, super cool that you had the initiative!

**Comment by lsmith77** [comment]
thx for the comments

FYI I was basing the plugin on https://github.com/PostHog/first-time-event-tracker

I will address your comments, ie. remove `Set()`, remove the `session_started` check. I will also add special handling for `ip` so that we only do `delete event.ip` if `$ip` is in the list of properties.

**Comment by lsmith77** [comment]
@yakkomajuri done

**Pull request merged**